pkg/vpc/cleanup.go

```diff
@@ -30,49 +33,50 @@ func findAvailableNetworkInterfaces(provider api.ClusterProvider, spec *api.Clus
 			},
 		},
 	}
-	output, err := provider.EC2().DescribeNetworkInterfaces(input)
-	if err != nil {
-		return nil, errors.Wrapf(err, "unable to list dangling network interfaces in %q", spec.VPC.ID)
-	}
+
 	securityGroupRE, err := regexp.Compile(fmtSecurityGroupNameRegexForCluster(spec.Metadata.Name))
 	if err != nil {
-		return nil, errors.Wrapf(err, "unable to list dangling network interfaces in %q", spec.VPC.ID)
+		return errors.Wrap(err, "failed to create security group regex")
 	}
-	networkInterfaces := []string{}
-	for _, eni := range output.NetworkInterfaces {
-		id := *eni.NetworkInterfaceId
-		for _, sg := range eni.Groups {
-			if securityGroupRE.MatchString(*sg.GroupName) {
-				logger.Debug("found %q, which belongs to our security group %q (%s)", id, *sg.GroupName, *sg.GroupId)
-				networkInterfaces = append(networkInterfaces, id)
-				break
-			} else {
+
+	var lastErr error
+
+	err = ec2API.DescribeNetworkInterfacesPagesWithContext(ctx, input, func(output *ec2.DescribeNetworkInterfacesOutput, lastPage bool) bool {
```

Comment:
Do you think we need to pass context? Take a look at these:
```
pkg/cfn/manager/api.go:	if err := c.provider.CloudFormation().ListStacksPages(input, pager); err != nil {
pkg/cfn/manager/api.go:	if err := c.provider.CloudFormation().DescribeStackEventsPages(input, pager); err != nil {
pkg/cfn/manager/api.go:	if err := c.provider.CloudTrail().LookupEventsPages(input, pager); err != nil {
```

Whichever way we decide, let's make sure we use the same pattern in all instances of paged requests :)